addons/binding/org.openhab.binding.pioneeravr/src/main/java/org/openhab/binding/pioneeravr/discovery/PioneerAvrDiscoveryParticipant.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.pioneeravr.discovery;
+
+import java.util.Dictionary;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.UpnpDiscoveryParticipant;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.jupnp.model.meta.RemoteDevice;
+import org.openhab.binding.pioneeravr.PioneerAvrBindingConstants;
+import org.osgi.service.cm.ConfigurationException;
+import org.osgi.service.cm.ManagedService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An UpnpDiscoveryParticipant which allows to discover Pioneer AVRs.
+ * 
+ * @author Antoine Besnard
+ *
+ */
+public class PioneerAvrDiscoveryParticipant implements UpnpDiscoveryParticipant, ManagedService {
+
+	private Logger logger = LoggerFactory.getLogger(PioneerAvrDiscoveryParticipant.class);
+
+	private boolean isAutoDiscoveryEnabled;
+	private Set<ThingTypeUID> supportedThingTypes;
+
+	public PioneerAvrDiscoveryParticipant() {
+		this.isAutoDiscoveryEnabled = true;
+		this.supportedThingTypes = new HashSet<>();
+	}
+
+	@Override
+	public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+		return supportedThingTypes;

```

Comment:
My point was that the default (i.e. if NO configuration is given at all) should be enabled as it is a rather unusual case to deactivate discovery (note that this is what the inbox is for: Discovery does not mean that it is added to the system, but it first needs to be approved).

Even with the current code, this is still not the case: As you do a check for ```componentContext.getProperties() != null```, the returned supportedThingTypes will be empty, although your default is ```isAutoDiscoveryEnabled = true```. You need to make sure that the supportedThingTypes are correctly filled in that case.